### PR TITLE
Fix #390.

### LINKED ISSUE #390: Rust functions and python functions have the same type
Rust functions should have a different type:

```
>>> type(abs)
<class 'builtin_function_or_method'>
```

This is important because certain methods don't make sense. (```__code__``` in particular.)

### PR DESCRIPTION
**Pull request by Tom1380**

### CODE REVIEW

File: vm/src/obj/objfunction.rs [hunk 1]
```diff
@@ -15,6 +15,19 @@ pub fn init(context: &PyContext) {
         context.new_member_descriptor(function_code),
     );
 
+    let builtin_function_or_method_type = &context.builtin_function_or_method_type();
```
Comment (cthulahoops) [inline vm/src/obj/objfunction.rs]: I don't think we need to call the method here, you can just access the member directly: we just need a reference, we don't need to clone it. (See function_type above lacks parens.)

This pattern of having a method and member of the same name where one does a clone is confusing, sorry.

File: vm/src/obj/objfunction.rs [hunk 2]
```diff
@@ -15,6 +15,19 @@ pub fn init(context: &PyContext) {
...
+    context.set_attr(
+        &builtin_function_or_method_type,
+        "__get__",
+        context.new_rustfunc(bind_method),
+    );
+
+    context.set_attr(
```
Comment (cthulahoops) [inline vm/src/obj/objfunction.rs]: ```builtin_function_or_method_type``` doesn't have a ```__code__```. (This difference is what led me to realise we needed a different type.)

Comment (Tom1380) [reply vm/src/obj/objfunction.rs]: Ok I'm working on it.

Comment (Tom1380) [reply vm/src/obj/objfunction.rs]: Ok the changes are done.

File: vm/src/obj/objfunction.rs (hunk 1 above)
Comment (Tom1380) [reply vm/src/obj/objfunction.rs]: I fixed this with 7e1589d.

File: vm/src/obj/objfunction.rs (hunk 2 above)
Comment (Tom1380) [reply vm/src/obj/objfunction.rs]: I fixed this with c04f421.

**Pull request merged**